src/Shape/Box.js

```diff
@@ -0,0 +1,267 @@
+
+import Line from "./Line";
+import Rect from "./Rect";
+import Whisker from "./Whisker";
+
+/**
+    @class Box
+    @extends BaseClass
+    @desc Creates SVG box based on an array of data.
+*/
+export default class Box extends BaseClass {
+
+  /**
+      @memberof Box
+      @desc Invoked when creating a new class instance, and overrides any default parameters inherited from BaseClass.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._id = accessor("id", 1);
+    this._name = "Box";
+    this._rectConfig = {
+      fill: constant("white"),
+      stroke: constant("black"),
+      strokeWidth: constant(1)
+    };
+    this._rectHeight = d => d.third - d.first;
+    this._rectWidth = constant(50);
+    this._value = accessor("value");
+    this._whiskerMode = ["extent", "extent"];
+    this._x = constant(250);
+    this._y = constant(250);
+
+  }
+
+  /**
+      @memberof Box
+      @desc Filters/manipulates the data array before binding each point to an SVG group.
+      @param {Array} [*data* = the data array to be filtered]
+      @private
+  */
+  _dataFilter(data) {
+    const boxes = this._box = nest().key(this._id).entries(data).map(d => {
+
+      d.data = merge(d.values);
+      const values = d.values.map(this._value);
+      values.sort((a, b) => a - b);
+      d.i = data.indexOf(d.values[0]);
+      d.x = this._x(d.data, d.i);
+      d.y = this._y(d.data, d.i);
+
+      d.first = quantile(values, 0.25);
+      d.median = quantile(values, 0.50);
+      d.third = quantile(values, 0.75);
+
+      d.width = this._rectWidth();
+      d.height = this._rectHeight(d);
+      
+      const mode = this._whiskerMode;
+
+      if (mode[0] === "tukey") d.top = d.first - (d.third - d.first) * 1.5;
+      else if (mode[0] === "extent") d.top = max(values);
+      else if (typeof mode[0] === "number") d.top = min([max(values), quantile(values, mode[0] / 100)]);
+
+      if (mode[1] === "tukey") d.bottom = d.third + (d.third - d.first) * 1.5;
+      else if (mode[1] === "extent") d.bottom = min(values);
+      else if (typeof mode[1] === "number") d.bottom = max([min(values), quantile(values, mode[1] / 100)]);
+
+      this._topWhiskerLength = d.top - d.third;
+      this._bottomWhiskerLength = d.first - d.bottom;
+
+      d.nested = true;
+      d.__d3plusShape__ = true;
+
+      return d;
+    });
+
+    boxes.key = d => d.key;
+    return boxes;
+
+  }
+
+  /**
+      @memberof Box
+      @desc Draws the Box.
+      @param {Function} [*callback*]
+      @chainable
+  */
+  render() {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const filteredData = this._dataFilter(this._data);
+
+    // Draw box.
+    new Rect()
+      .data(filteredData)
+      .x(filteredData[0].x)
+      .y(filteredData[0].y)
```

Comment:
these `x` and `y` accessors are causing all of the boxes to have the same position! remove them